typescript/src/lib/utils/ledger/ledger.ts

```diff
@@ -0,0 +1,170 @@
+import { ethers } from "ethers"
+import {
+  Account,
+  WalletAPIClient,
+  WindowMessageTransport,
+} from "@ledgerhq/wallet-api-client"
+import BigNumber from "bignumber.js"
+import { Hex } from "../../utils"
+import { AddressZero } from "@ethersproject/constants"
+import { Deferrable } from "@ethersproject/properties"
+import { getWalletAPIClient, getWindowMessageTransport } from "."
+import { Signer } from "ethers"
+
+export class LedgerLiveAppEthereumSigner extends Signer {
```

Comment:
```suggestion
export class LedgerLiveEthereumSigner extends Signer {
```

Reply:
I wanted to strictly clarify that this is a SIgner used in Live App inside Ledger Live. This is because there are other ways of connecting a ledger device that also uses Ledger Live. For example you can use [dapp-connect-kit](https://developers.ledger.com/docs/dapp-connect-kit/introduction/) to connect your account to the dapp using the Ledger Live app (notice the lowercase letter for "app"), but in this case the signer is strictly for a [Live App ](https://developers.ledger.com/docs/live-app/start-here/) (notice the uppercase letter in "App". "Live App" in this case is the full name for the app that is embeded and displayed in the `Discover` section in the Ledger Live application).

I don't mind changing it though so let me know what you think.

Reply:
I'd simplify the name to `LedgerLiveEthereumSigner` and write a good docstring for it explaining the purpose of this component.

Reply:
8244f9f01aaa2cfd1dcefc36bc716e824230fe57